src/jit/hwintrinsiclistxarch.h

```diff
@@ -11,59 +11,80 @@
 // clang-format off
 
 #if FEATURE_HW_INTRINSICS
-//                  Intrinsic ID                   Function name                ISA   
+/* Note
+    1) Each hardware intrinsic has a unique Intrinsic ID with type of `enum NamedIntrinsic`
+    2) All the overloads of an intrinsic in an ISA class share one Intrinsic ID
+    3) The intrinsic that generates instructions with a fixed imm8 operand has a `ival` field with "not -1" value, e.g., Sse.CompareEqual(v1,v2) -> cmpps xmm0, xmm1, 0
+    4) SIMD intrinsics have a non-zero `SIMD size` field based-on that operate over `Vector128<T>` (16) or `Vector256<T>`
+    5) Scalar intrinsics that operate over general purpose registers (e.g., Sse41.Crc32) have `SIMD size` with 0
+    6) Each intrinsic has a `NumArg` for number of parameters, and some intrinsics that are overloaded on multiple parameter numbers have this field with -1
+    7) Each intrinsic has 10 `instructions` fields that list the instructions should be generated based-on the base type
+    8) Each intrinsic has one or more category with type of `enum HWIntrinsicCategory`, please see the definition of HWIntrinsicCategory for details
+*/
+// **************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
+//                  Intrinsic ID                                      Function name                                      ISA        ival       SIMD size       NumArg                                                                   instructions                                                                                       Category
+//                                                                                                                                                                            {TYP_BYTE,      TYP_UBYTE,     TYP_SHORT,     TYP_USHORT,    TYP_INT,       TYP_UINT,      TYP_LONG,      TYP_ULONG,     TYP_FLOAT,     TYP_DOUBLE}
+// **************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************************
 //  SSE Intrinsics          
```

Comment:
There is quite a bit of trailing white space in this file (at least the one I have fetched from github)